lib/geolocation.js

```diff
@@ -0,0 +1,46 @@
+function Geolocation() {}
+
+Geolocation.prototype = {
+
+  isSupport: function() {
+    return this.supportsGeolocation;
+  },
+
+  getCurrentPosition: function() {
+    const positionOptions = {
+      enableHighAccuracy: true
+    };
+
+    return new Promise(function(res, rej) {
+      window.navigator.geolocation.getCurrentPosition(res, rej, positionOptions);
+    });
+  },
+
+  checkGeolocationSupport: function () {
+    return new Promise(function(res, rej) {
+      if (this.supportsGeolocation !== undefined) {
+        res(this.supportsGeolocation);
+    
+      } else if (window.navigator.permissions !== undefined) {
+        // navigator.permissions has incomplete browser support
+        // http://caniuse.com/#feat=permissions-api
+        // Test for the case where a browser disables Geolocation because of an
+        // insecure origin
+        window.navigator.permissions.query({name: 'geolocation'}).then(function (p) {
+          //TODO: PermissionStatus{state: 'denied', onchange: null}
```

Comment:
What are your thoughts on messaging the user when navigator permission is denied?

Reply:
User will see "You denied Geolocation".

<img width="290" alt="Screenshot 2021-12-08 at 15 33 09" src="https://user-images.githubusercontent.com/6375386/145209000-4d4bc721-ae5b-4eb8-aa93-190920f07cee.png">

Reply:
@krydima @mapbox-danny I'm proposing we change the text to say "Geolocation permission denied"